pkg/neutronapi/deployment.go

```diff
@@ -0,0 +1,148 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package neutronapi
+
+import (
+	common "github.com/openstack-k8s-operators/lib-common/pkg/common"
+	neutronv1 "github.com/openstack-k8s-operators/neutron-operator/api/v1beta1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	// ServiceCommand -
+	ServiceCommand = "/usr/local/bin/kolla_set_configs && /usr/local/bin/kolla_start"
+)
+
+// Deployment func
+func Deployment(
+	instance *neutronv1.NeutronAPI,
+	configHash string,
+	labels map[string]string,
+) *appsv1.Deployment {
+	runAsUser := int64(0)
+
+	livenessProbe := &corev1.Probe{
+		// TODO might need tuning
+		TimeoutSeconds:      5,
+		PeriodSeconds:       3,
+		InitialDelaySeconds: 3,
+	}
+	readinessProbe := &corev1.Probe{
+		// TODO might need tuning
+		TimeoutSeconds:      5,
+		PeriodSeconds:       5,
+		InitialDelaySeconds: 5,
+	}
+
+	args := []string{"-c"}
+	if instance.Spec.Debug.Service {
+		args = append(args, common.DebugCommand)
+		livenessProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/bin/true",
+			},
+		}
+
+		readinessProbe.Exec = &corev1.ExecAction{
+			Command: []string{
+				"/bin/true",
+			},
+		}
+	} else {
+		args = append(args, ServiceCommand)
+		//
+		// https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+		//
+		livenessProbe.HTTPGet = &corev1.HTTPGetAction{
+			Path: "/",
+			Port: intstr.IntOrString{Type: intstr.Int, IntVal: int32(NeutronPublicPort)},
+		}
+		readinessProbe.HTTPGet = &corev1.HTTPGetAction{
+			Path: "/",
+			Port: intstr.IntOrString{Type: intstr.Int, IntVal: int32(NeutronPublicPort)},
+		}
```

Comment:
One thing to check here: do these `/` queries return a 200 response code?  If not, they will generate warnings every few seconds.  We had to change to a different path for Glance, for instance: https://github.com/openstack-k8s-operators/glance-operator/blob/0df23ef3490a6e0ab83f66fa355174d2e11bef0b/pkg/glance/deployment.go#L73-L80

Reply:
Yes it returned 200 response so i used that, but yes can check if some better path can be used instead.